src/main/java/de/fraunhofer/aisec/cpg/passes/JavaExternalTypeHierarchyResolver.java

```diff
@@ -0,0 +1,56 @@
+package de.fraunhofer.aisec.cpg.passes;
+
+import com.github.javaparser.resolution.UnsolvedSymbolException;
+import com.github.javaparser.resolution.declarations.ResolvedReferenceTypeDeclaration;
+import com.github.javaparser.resolution.types.ResolvedReferenceType;
+import com.github.javaparser.symbolsolver.model.resolution.SymbolReference;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+import de.fraunhofer.aisec.cpg.TranslationResult;
+import de.fraunhofer.aisec.cpg.frontends.java.JavaLanguageFrontend;
+import de.fraunhofer.aisec.cpg.graph.TypeManager;
+import de.fraunhofer.aisec.cpg.graph.type.Type;
+import de.fraunhofer.aisec.cpg.graph.type.TypeParser;
+import java.util.HashSet;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JavaExternalTypeHierarchyResolver extends Pass {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(JavaExternalTypeHierarchyResolver.class);
+
+  @Override
+  public void accept(TranslationResult translationResult) {
+    // Run only for Java.
+    if (JavaLanguageFrontend.class.equals(this.lang.getClass())) {
```

Comment:
Could we add function to the `Pass` interface that returns the list of supported frontends? We probably need more passes that are language specific, i.e. the upcoming C++-only pass that connects definitions and declarations. We could already prepare this here.